### ISSUE TITLE
Not compatible with flutter_stripe 3.1.0 because of freezed_annotation

### ISSUE THREAD

**Issue by mayouf** [RevenueCat/purchases-flutter#387]
When I use the last version of the plugin `flutter_stripe ^3.1.0` I get this message in console:
```
Because flutter_stripe >=3.1.0 depends on stripe_platform_interface ^3.1.0 which depends on freezed_annotation ^2.0.3, flutter_stripe >=3.1.0 requires freezed_annotation ^2.0.3.
And because purchases_flutter 3.10.0 depends on freezed_annotation ^1.1.0 and no versions of purchases_flutter match >3.10.0 <4.0.0, flutter_stripe >=3.1.0 is incompatible with purchases_flutter ^3.10.0.
So, because XXXXXX depends on both purchases_flutter ^3.10.0 and flutter_stripe ^3.1.0, version solving failed.
pub get failed (1; So, because XXXXXX depends on both purchases_flutter ^3.10.0 and flutter_stripe ^3.1.0, version solving failed.)
Process finished with exit code 1
```

Seems like `purchases_flutter` does not use the last version of `freezed_annotation`

**Comment by aboedo** [comment]
Hi @mayouf! Thanks for opening. 
This is a duplicate of https://github.com/RevenueCat/purchases-flutter/issues/375, and this [answer from that issue](https://github.com/RevenueCat/purchases-flutter/issues/375#issuecomment-1144071738) still holds true: we've updated to freezed 2, and we're going to ship it as part of our upcoming 4.0.0 release. 
Let me know if you have any questions!!

**Comment by mayouf** [comment]
THX 🙂

**Comment by aboedo** [comment]
@mayouf thank you for your patience on this!

**Comment by vegaro** [comment]
Just an fyi. We published a pre-release `4.0.0-beta.1` to pub.dev. We'll be publishing the final version soon. This other issue https://github.com/RevenueCat/purchases-flutter/issues/391 also has a conversation about the new version.
